sycl/doc/extensions/experimental/sycl_ext_oneapi_complex.asciidoc

```diff
@@ -0,0 +1,334 @@
+    template <class T> constexpr T real(const complex<T> &);
+    /// Returns the real component of the number y, treated as complex numbers with zero imaginary component.
+    template <class T> constexpr T real(T);
+    /// Returns the imaginary component of the complex number z.
+    template <class T> constexpr T imag(const complex<T> &);
+    /// Returns the imaginary component of the number y, treated as complex numbers with zero imaginary component.
+    template <class T> constexpr T imag(T);
+
+    /// Compute the magnitude of complex number x.
+    template <class T> T abs(const complex<T> &);
+    /// Compute phase angle in radians of complex number x.
+    template <class T> T arg(const complex<T> &);
+    /// Compute phase angle in radians of complex number x, treated as complex number with positive zero imaginary component.
+    template <class T> T arg(T);
+    /// Compute the squared magnitude of complex number x.
+    template <class T> T norm(const complex<T> &);
+    /// Compute the squared magnitude of number x, treated as complex number with positive zero imaginary component.
+    template <class T> T norm(T);
+    /// Compute the conjugate of complex number x.
+    template <class T> complex<T> conj(const complex<T> &);
+    /// Compute the conjugate of number y, treated as complex number with positive zero imaginary component.
+    template <class T> complex<T> conj(T);
+    /// Compute the projection of complex number x.
+    template <class T> complex<T> proj(const complex<T> &);
+    /// Compute the projection of number y, treated as complex number with positive zero imaginary component.
+    template <class T> complex<T> proj(T);
+    /// Construct a complex number from polar coordinates with mangitude rho and angle theta.
+    template <class T> complex<T> polar(const T &, const T &);
+
+    /// TRANSCENDENTALS:
+    /// Compute the natural log of complex number x.
+    template <class T> complex<T> log(const complex<T> &);
+    /// Compute the base-10 log of complex number x.
+    template <class T> complex<T> log10(const complex<T> &);
+    /// Compute the square root of complex number x.
+    template <class T> complex<T> sqrt(const complex<T> &);
+    /// Compute the base-e exponent of complex number x.
+    template <class T> complex<T> exp(const complex<T> &);
+
+    /// Compute complex number z raised to the power of complex number y.
+    template <class T> complex<T> pow(const complex<T> &, const complex<T> &);
+    /// Compute complex number z raised to the power of complex number y.
+    template <class T, class U> complex</*Promoted*/> pow(const complex<T> &, const complex<U> &);
+    /// Compute complex number z raised to the power of real number y.
+    template <class T, class U> complex</*Promoted*/> pow(const complex<T> &, const U &);
+    /// Compute real number x raised to the power of complex number y.
+    template <class T, class U> complex</*Promoted*/> pow(const T &, const complex<U> &);
```

Comment:
I think it would be worth explaining what the return type is in these cases that show "Promoted".  The rules are not exactly the same as standard C++ because you have the `half` type.  I think you would follow the wording for "until C++23" here:

https://en.cppreference.com/w/cpp/numeric/complex/pow

Except you would remove the stuff about `long double` and add wording for `half`.